Makefile

```diff
@@ -69,12 +69,13 @@ test-merge:
 	if test -n "$$REV"; then \
 	  FILES=$$(git diff --name-only FETCH_HEAD `git merge-base FETCH_HEAD master 2>/dev/null` | grep 'tests.*pm') ;\
 	  for file in $$FILES; do if test -f $$file; then \
-	    tools/check_metadata $$file || touch failed; \
-	    git grep -q wait_idle $$file && touch failed; \
-	    ${PERLCRITIC} $$file || (echo $$file ; touch failed) ;\
+	    tools/check_metadata $$file || touch failed1; \
+	    git grep -q wait_idle $$file && echo $$file >>failed2; \
```

Comment:
I guess you can simply output the file here instead of delaying that (in the rather complicated way) to later.

Would be great if we could simply reuse the make targets as defined above with the "-changed" suffix, e.g. as an optional argument. If "reusing a make target" is not possible we can define the commands within a Makefile variable and use the command here and in the make targets that check all the files.

Reply:
First, if I simply output the file, users might can't understand what mistake happen. 

About reuse the make targets as defined above with the "changed" suffix,
check_metadata script needs an argument so line 62~64 is wrong. because when user commits modification on !".pm" files, this checking go failed.  but here, due to FOR loop will filter NULL , so we needn't care here.

Reply:
Our checking only working for ".pm" files ? If we also care !".pm" files, if we ONLY care ".pm" files, I could modify "check_metadata" script turn to an "optional argument".

Reply:
yes, it's ok if the "check_metadata" is only executed for "\.pm" files, actually even only "tests/.*\.pm" would be enough.

Reply:
@okurz please review the following comments.